sync-android/src/main/java/com/cloudant/sync/sqlite/android/encryption/AttachmentEncryptionUtils.java

```diff
@@ -0,0 +1,274 @@
+                throw new SecurityUtilsException("The encrypted object was generated in another environment. Cannot decrypt in this environment.");
+            }
+
+            cipherText = encryptedObject.getString(CYPHER_TEXT_LABEL);
+            String iv = encryptedObject.getString(IV_LABEL);
+            */
+            byte[] bytes = hexStringToByteArray(cipherText);
+
+            Cipher cipher = initCipher(Cipher.DECRYPT_MODE, hexStringToByteArray(key), iv);
+
+            byte[] original = cipher.doFinal(bytes);
+
+            return new String(original, UTF_8);
+        } catch (Exception e) {
+            throw new SecurityUtilsException("There was a problem while adding properties to the returned JSONObject.", e);
+        }
+    }
+
+    /**
+     * Generate a hexadecimal-encoded random byte string locally with the given byte length.
+     * @param byteLength the length of the string to be generated, in byte size
+     * @return hexadecimal-encoded string of the generated random string
+     */
+    public static String getRandomString(int byteLength){
+        byte[] randomByteArray = generateLocalKey(byteLength);
+        return encodeBytesAsHexString(randomByteArray);
+    }
+
+    public static final byte[] hexStringToByteArray(String s) {
+        int len = s.length();
+        byte[] data = new byte[len / 2];
+        for (int i = 0; i < len; i += 2) {
+            data[i / 2] = (byte) ((Character.digit(s.charAt(i), 16) << 4)
+                    + Character.digit(s.charAt(i+1), 16));
+        }
+        return data;
+    }
+
+    //Called if the user requests to generate the key locally, we gen all 256 bits,
+    //so no need to use PBKDF2 like the method above
+    public static byte[] generateLocalKey(int numBytes){
+        byte[] randBytes = new byte[numBytes];
+        new SecureRandom().nextBytes (randBytes);
+        return randBytes;
+    }
+
+    public static String encodeBytesAsHexString (byte bytes[]) {
+        StringBuilder result = new StringBuilder();
+
+        if (bytes != null) {
+            for (byte curByte : bytes) {
+                result.append (String.format ("%02X", curByte)); //$NON-NLS-1$
+            }
+        }
+
+        return result.toString();
+    }
+
+
+    //TODO Encryption/cipher algorithm and details should be same as key management
+    static Cipher initCipher(int mode, byte[] key, byte[] iv) throws NoSuchAlgorithmException, NoSuchPaddingException, InvalidKeyException, InvalidAlgorithmParameterException {
+
+        IvParameterSpec ivspec = new IvParameterSpec(iv);
+        SecretKeySpec skeySpec = new SecretKeySpec(key, "AES"); //$NON-NLS-1$
+
+        // initialize the cipher for encrypt mode
+        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding"); //$NON-NLS-1$
+        cipher.init(mode, skeySpec, ivspec);
+
+        return cipher;
+    }
+
+    public static final String byteArrayToHexString(byte[] bytes) {
```

Comment:
This is a generic utility not specific to crypto, and also may be covered by the apache codec classes. I think @mikerhodes was contributing something similar to the 43715 so you should have a look there and ask him.